notification/slack/channel.go

```diff
@@ -327,30 +331,52 @@ func (s *ChannelSender) Send(ctx context.Context, msg notification.Message) (*no
 			break
 		}
 
-		vals.Set("text", fmt.Sprintf("Alert: %s\n\n<%s>", t.Summary, cfg.CallbackURL("/alerts/"+strconv.Itoa(t.AlertID))))
+		a.Summary = t.Summary
+		a.ID = t.AlertID
+		a.Status = alert.StatusTriggered
+
+		msgOpt := CraftAlertMessage(a, cfg.CallbackURL("/alerts/"+strconv.Itoa(a.ID)), "")
+		_, ts, _, err := api.SendMessageContext(ctx, msg.Destination().Value, msgOpt...)
+		if err != nil {
+			return nil, err
+		}
+		return &notification.SentMessage{
+			ExternalID: ts,
+			State:      notification.StateDelivered,
+			SrcValue:   "",
+		}, nil
 	case notification.AlertStatus:
-		vals.Set("thread_ts", t.OriginalStatus.ProviderMessageID.ExternalID)
-		var status string
-		switch t.NewAlertStatus {
-		case alert.StatusActive:
-			status = "Acknowledged"
-		case alert.StatusTriggered:
-			status = "Unacknowledged"
-		case alert.StatusClosed:
-			status = "Closed"
+		_a, err := s.cfg.AlertStore.FindOne(ctx, t.AlertID)
+		if err != nil {
+			return nil, err
 		}
+		a = *_a
 
-		text := "Status Update: " + status + "\n" + t.LogEntry
-		vals.Set("text", text)
+		msgOpt := CraftAlertMessage(a, cfg.CallbackURL("/alerts/"+strconv.Itoa(a.ID)), "")
+		_, ts, _, err := api.UpdateMessageContext(ctx, msg.Destination().Value, t.Dest.Value, msgOpt...)
+		if err != nil {
+			return nil, err
+		}
+		vals.Set("thread_ts", t.OriginalStatus.ProviderMessageID.ExternalID)
+		// todo, post to thread
+		return &notification.SentMessage{
+			ExternalID: ts,
+			State:      notification.StateDelivered,
+			SrcValue:   "",
+		}, nil
 	case notification.AlertBundle:
+		vals.Set("token", cfg.Slack.AccessToken)
 		vals.Set("text", fmt.Sprintf("Service '%s' has %d unacknowledged alerts.\n\n<%s>", t.ServiceName, t.Count, cfg.CallbackURL("/services/"+t.ServiceID+"/alerts")))
+		return post(s, vals)
 	case notification.ScheduleOnCallUsers:
 		vals.Set("text", s.onCallNotificationText(ctx, t))
-	default:
-		return nil, errors.Errorf("unsupported message type: %T", t)
+		return post(s, vals)
 	}
-	vals.Set("token", cfg.Slack.AccessToken)
 
+	return nil, errors.Errorf("unsupported message type: %T", msg.Type())
+}
+
+func post(s *ChannelSender, vals url.Values) (*notification.SentMessage, error) {
```

Comment:
Should we just use the api library for this? we should also call it `chatPostMessage` or something more clear

Reply:
i left this here until i wrap up the rest of the logic so all send routes still work. but yeah i can essentially do that to send a text Slack message for bundling and on-call user notifications

Reply:
completed- all actions should use the slack-go lib now